tensorflow_datasets/core/features/image_feature.py

```diff
@@ -345,3 +370,46 @@ def _get_and_validate_shape(shape, encoding_format):
         f'{acceptable_channels} (was {channels})'
     )
   return tuple(shape)
+
+
+def _get_and_validate_colormap(use_colormap, shape, encoding_format):
+  """Update the shape."""
+  if use_colormap and encoding_format and encoding_format != 'png':
+    raise ValueError(
+        f'Colormap is only available for PNG images. Not {encoding_format}'
+    )
+  if shape[-1] != 1 and use_colormap:
+    raise ValueError('Colormap is only available for gray-scale images')
+
+  return use_colormap
+
+
+@utils.memoize()
+def _get_colormap(cmap_size: int) -> np.ndarray:
+  """Generate cmap configuration"""
+  cmap_list = []
+  cmap_name = 'hsv'
+  matplotlib = lazy_imports_lib.lazy_imports.matplotlib  # pylint: disable=invalid-name
+
+  # cmap returned in the rgba format
+  cmap = matplotlib.cm.get_cmap(cmap_name, cmap_size)
+  for itr in range(cmap_size):
+    cmap_list.append(cmap(itr))
+
+  return cmap_list
+
+
+def _apply_colormap(image: np.ndarray) -> np.ndarray:
+  """Apply colormap to 1D images"""
+  image = image.squeeze(axis=-1)
+  unique_pixels = np.unique(image)
+  colors = _get_colormap(len(unique_pixels))
```

Comment:
The number of pixels might change from images to images, but we want to colors to always be the same across images. We should generate a unique colormap of 255 values.

Reply:
Generating a colormap for soo many values would lead to non-unique shades (at least to the human eye) ( or I maybe wrong here )
Is this something we want? If not maybe we could have separate mapping zones ( 1-20 shades, 1-50 shades, and .....  this would mean only those with categories>~20 will suffer from non-unique looking shades )

![image](https://user-images.githubusercontent.com/23381512/105839569-fd746780-5fd1-11eb-8b17-781eb17c5312.png)
![image](https://user-images.githubusercontent.com/23381512/105839594-0a915680-5fd2-11eb-9b24-99c41c7f54a5.png)
![image](https://user-images.githubusercontent.com/23381512/105839639-1c72f980-5fd2-11eb-9bff-71f96423117b.png)
![image](https://user-images.githubusercontent.com/23381512/105839697-301e6000-5fd2-11eb-97da-d3c182253c8d.png)

Reply:
This is a good point. Basically what we want is that numbers close to each others have high contrast. But indeed, many values will have similar colors.
So it would be something like 

```
0 -> black
1 -> red
2 -> green
3 -> blue
...
123 -> slight variation on blue
124 -> slight variation on red
...
```

So an image which has contains values `[1, 3, 0]` would have distinct colors, but an image with  values [3, 123] would have very low contrast (as 3 and 123 correspond to similar colors in the above mapping). In practice it wouldn't really matter as the likelihood that `[3, 123]` appear on the same image is low.

But more important, because the same colors are applied on all images, so value `x` will always be displayed the same across all images.

One way to create the color map would be randomly shuffle the values (using fixed seed for determinism). Another likely smarter way would be to reorder the mapping returned by matplotlib to better distribute the values (to have something more like my example mapping above)

Another based color could be used too, like `gist_ncar` which seems to have more color variations

Reply:
Here is the contrast of the colormaps, feels like only the order has changed for the colors.
![image](https://user-images.githubusercontent.com/23381512/105885175-11878b80-6009-11eb-8cf5-476d13ad7c87.png)
![image](https://user-images.githubusercontent.com/23381512/105885182-14827c00-6009-11eb-886c-e02542cf4773.png)

I believe our motive is to assign deterministic colors for pixel values where any pixel from any image ( from the same dataset ) will have the same colormap. Is this understanding correct? If that's the case, we can use some metadata like  `dataset_name` as the seed to generate uniform color distribution from the large color space ( 255 values ), this would ensure that the colors are distinguishable. What do you think about this implementation?

Reply:
Let me think of an algorithm to generate the colormap as we want and I'll post it there maybe tomorrow.

In the meantime, could you assume `_get_colormap()` returns a `[255, 3]` vector (you can have `_get_colormap` return a dummy `range(255*3).reshape(255,3)` for now) and add the update the tests ?